src/stdio/ScalarIO.sac

```diff
@@ -192,130 +192,153 @@ inline void fprint(TermFile &stream, bool n, int mode)
-SHOW_TERMFILE_S_(longlong)
-SHOW_TERMFILE_S_(ushort)
-SHOW_TERMFILE_S_(ubyte)
-SHOW_TERMFILE_S_(uint)
-SHOW_TERMFILE_S_(ulong)
-SHOW_TERMFILE_S_(ulonglong)
+SHOW_TERMFILE_S_(short, hd)
+SHOW_TERMFILE_S_(byte, x)
+SHOW_TERMFILE_S_(longlong, lld)
+SHOW_TERMFILE_S_(ushort, hu)
+SHOW_TERMFILE_S_(ubyte, X)
+SHOW_TERMFILE_S_(uint, u)
+SHOW_TERMFILE_S_(ulong, lu)
+SHOW_TERMFILE_S_(ulonglong, llu)
 #endif /* FULLTYPES */
 
-#define PRINT_TERMFILE_S_(typ, format)               \
-inline void print( typ n)                            \
-{                                                    \
-  TermFile::printf("Dimension:  0\nShape    : < >\n");\
-  TermFile::printf( " " QUOTE(%format\n), n);        \
-}
-
-PRINT_TERMFILE_S_(int, d)
-PRINT_TERMFILE_S_(long, ld)
-PRINT_TERMFILE_S_(string, s)
-
-#ifdef FULLTYPES
-PRINT_TERMFILE_S_(short, hd)
-PRINT_TERMFILE_S_(byte, x)
-PRINT_TERMFILE_S_(longlong, lld)
-PRINT_TERMFILE_S_(ushort, hu)
-PRINT_TERMFILE_S_(ubyte, X)
-PRINT_TERMFILE_S_(uint, u)
-PRINT_TERMFILE_S_(ulong, lu)
-PRINT_TERMFILE_S_(ulonglong, llu)
-#endif /* FULLTYPES */
-
-void print(char c)
+/* special definitions of show for char, float, double, and bool */
+inline void show(char n)
 {
-  TermFile::printf("Dimension:  0\nShape    : < >\n");
-  TermFile::printf(" '%c'\n", c);
+  TermFile::printf("'%c'\n", n);
 }
 
-inline void print(float n)
+inline void show(float n)
 {
-  TermFile::printf("Dimension:  0\nShape    : < >\n");
-  TermFile::printf(" %g\n", tod(n));
+  TermFile::printf("%g\n", tod(n));
 }
 
-inline void print(float n, int prec)
+inline void show(float n, int prec)
 {
-  TermFile::printf("Dimension:  0\nShape    : < >\n");
-  TermFile::printf(" %.*g\n", prec, tod(n));
+  TermFile::printf("%.*g\n", prec, tod(n));
 }
 
-inline void print(double n)
+inline void show(double n)
 {
-  TermFile::printf("Dimension:  0\nShape    : < >\n");
-  TermFile::printf(" %g\n", n);
+  TermFile::printf("%g\n", n);
 }
 
-inline void print(double n, int prec)
+inline void show(double n, int prec)
 {
-  TermFile::printf("Dimension:  0\nShape    : < >\n");
-  TermFile::printf(" %.*g\n", prec, n);
+  TermFile::printf("%.*g\n", prec, n);
 }
 
-inline void print(bool n)
+inline void show(bool n)
 {
-  TermFile::printf("Dimension:  0\nShape    : < >\n");
-  TermFile::printf(" %s\n", n ? "true" : "false");
+  TermFile::printf("%s\n", n ? "true" : "false");
 }
 
-inline void print(bool n, int mode)
+inline void show(bool n, int mode)
 {
-  TermFile::printf("Dimension:  0\nShape    : < >\n");
   if (n)
   {
-     if (mode==2)
-     {
-	TermFile::printf(" true\n");
-     }
-     else
-     {
-       if (mode==1)
-       {
-	 TermFile::printf(" t\n");
-       }
-       else
-       {
-	 TermFile::printf(" 1\n");
-       }
-     }
+    if (mode==2)
+    {
+      TermFile::printf("true\n");
+    }
+    else
+    {
+      if (mode==1)
+      {
+        TermFile::printf("t\n");
+      }
+      else
+      {
+        TermFile::printf("1\n");
+      }
+    }
```

Comment:
Maybe something like:
```c
TermFile::printf ("%s\n", mode == 2 ? "true" : mode == 1 ? "t" : "1");
```

Reply:
That doesn't look like any code I would ever write. I hate IF/THEN, but I hate nested ones even more.
Your proposed "something like" looks better to me.